job-service/src/main/java/com/hpe/caf/services/job/api/DatabaseHelper.java

```diff
@@ -591,4 +597,42 @@ private static void throwIfUnexpectedException(final SQLException se) throws Exc
             throw se;
         }
     }
+
+    private void buildGetJobsStatement(final String partitionId, String jobIdStartsWith, String statusType, Integer limit, Integer offset,
+                                       final SortField sortField, final SortDirection sortDirection, final List<String> labels,
+                                       final String filter, final CallableStatement stmt, final Connection conn) throws Exception
+    {
+        if (jobIdStartsWith == null) {
+            jobIdStartsWith = "";
+        }
+        if (statusType == null) {
+            statusType = "";
+        }
+        if (limit == null) {
+            limit = 0;
+        }
+        if (offset == null) {
+            offset = 0;
+        }
+        stmt.setString(1, partitionId);
+        stmt.setString(2, jobIdStartsWith);
+        stmt.setString(3, statusType);
+        stmt.setInt(4, limit);
+        stmt.setInt(5, offset);
+        stmt.setString(6, sortField.getDbField());
+        stmt.setString(7, sortField.getSortLabel());
+        stmt.setBoolean(8, sortDirection.getDbValue());
+        Array array;
+        if (labels != null) {
+            array = conn.createArrayOf("VARCHAR", labels.toArray());
+        } else {
+            array = conn.createArrayOf("VARCHAR", new String[0]);
+        }
+        stmt.setArray(9, array);
+        stmt.setString(10, filter);
+
+        if (array != null) {
```

Comment:
I'm surprised this doesn't crash, because we are free'ing the array before we execute the statement.